cpu/stm32/cpu_init.c

```diff
@@ -372,3 +372,43 @@ void cpu_init(void)
         _wlx5xx_init_subghz_debug_pins();
     }
 }
+
+void backup_ram_init(void)
+{
+/* see reference manual "Battery backup domain" */
+#if defined(RCC_APB1ENR_PWREN)
+    periph_clk_en(APB1, RCC_APB1ENR_PWREN);
+#elif defined(RCC_APBENR1_PWREN)
+    periph_clk_en(APB1, RCC_APBENR1_PWREN);
+#elif defined(RCC_APB1ENR1_PWREN)
+    periph_clk_en(APB1, RCC_APB1ENR1_PWREN);
+#elif defined(RCC_AHB3ENR_PWREN)
+    periph_clk_en(AHB3, RCC_AHB3ENR_PWREN);
+#endif
+    stmclk_dbp_unlock();
+#if defined(RCC_AHB1ENR_BKPSRAMEN)
+    periph_clk_en(AHB1, RCC_AHB1ENR_BKPSRAMEN);
+#endif
+}
+
+#ifndef BACKUP_RAM_MAGIC
+#define BACKUP_RAM_MAGIC    {'R', 'I', 'O', 'T'}
+#endif
+
+bool cpu_woke_from_backup(void)
+{
+#if IS_ACTIVE(CPU_HAS_BACKUP_RAM)
+    static const char _signature[] BACKUP_RAM_DATA = BACKUP_RAM_MAGIC;
+    /* switch off regulator to save power */
+#ifndef RIOTBOOT
+    pm_backup_regulator_off();
+#endif
```

Comment:
Is this supposed to be unavailable in `riotboot.elf`? However, `BOARD=nucleo-f767zi make -C tests/riotboot flash test` prints "TEST PASSED".

Reply:
This is only about saving a bit of energy right?
Given that riotboot only runs for a very brief moment and we want to keep it as small as possible, I'd say it's the right thing to do to leave it on here.

Of course we should not end up in a situation where using riotboot causes us to lose the contents of the backup RAM.

Reply:
Yes turning it off is just to save some power.
When I do `FEATURES_REQUIRED+=riotboot BOARD=nucleo-f446re make -c tests/periph_backup_ram flash term`, the counter in backup RAM is not reset, so I´d say it´s fine.